.buildkite/pipelines/pipeline.kibana-release.yaml

```diff
@@ -0,0 +1,9 @@
+steps:
+  - label: ":releasethekraken: Release kibana"
+    # https://regex101.com/r/tY52jo/1
+    if: 'BUILDKITE_TAG =~ "/^deploy@\d+$$/"'
```

Comment:
For this if, wouldn't we need `build.env()`? 

From: https://buildkite.com/docs/pipelines/conditionals
<img width="728" alt="Screenshot 2023-08-29 at 17 40 04" src="https://github.com/elastic/kibana/assets/4738868/9407baae-52ec-4495-b242-4b4d3090420f">